addons/mail/models/mail_thread.py

```diff
@@ -1146,7 +1166,17 @@ def _message_route_process(self, message, message_dict, routes):
             else:
                 # if a new thread is created, parent is irrelevant
                 message_dict.pop('parent_id', None)
-                thread = ModelCtx.message_new(message_dict, custom_values)
+                # Report failure/record success of message creation except if alias is not defined (fallback model case)
+                try:
+                    thread = ModelCtx.message_new(message_dict, custom_values)
+                except Exception:
+                    if alias:
+                        with self.pool.cursor() as new_cr:
```

Comment:
@rco-odoo Just to be sure, is it correct for you ? Rationale: when dealing with incoming emails, if we have an issue with the alias (like badly configured), we notify some people (send an email) there is an issue before letting propagate the error (and rollback). As we are in a transient state before the rollback, we use a new cursor for that purpose.